agent/agent_test.go

```diff
@@ -3433,6 +3433,20 @@ func TestAgent_ReloadConfigOutgoingRPCConfig(t *testing.T) {
 	require.Len(t, tlsConf.ClientCAs.Subjects(), 2)
 }
 
+func TestAgent_GetRLimits(t *testing.T) {
```

Comment:
Now that `checkLimitsFromMaxConnsPerClient` and its caller are both in `agent/config` it would be good to have the test which covers that behaviour also in the `agent/config` package. 

I think this test should be moved into the `agent/config` package. I guess the name will have to change as well, to something like `TestBuildAndValidate_HTTPMaxConnsPerClientExceedsRLimit`.